NIFI-10217 Refactor Registry Spring Security Configuration

exceptionfactory opened the pull request:
# Summary

[NIFI-10217](https://issues.apache.org/jira/browse/NIFI-10217) Refactors the NiFi Registry Spring Security Configuration, removing extension of the deprecated `WebSecurityConfigurerAdapter` class.

The updated configuration creates a `SecurityFilterChain` bean using the `HttpSecurity` builder and creates an `AuthenticationManager` using the existing Authentication Providers in the current configuration.

# Tracking

Please complete the following tracking steps prior to pull request creation.

### Issue Tracking

- [X] [Apache NiFi Jira](https://issues.apache.org/jira/browse/NIFI) issue created

### Pull Request Tracking

- [X] Pull Request title starts with Apache NiFi Jira issue number, such as `NIFI-00000`
- [X] Pull Request commit message starts with Apache NiFi Jira issue number, as such `NIFI-00000`

### Pull Request Formatting

- [X] Pull Request based on current revision of the `main` branch
- [X] Pull Request refers to a feature branch with one commit containing changes

# Verification

Please indicate the verification steps performed prior to pull request creation.

### Build

- [X] Build completed using `mvn clean install -P contrib-check`
  - [X] JDK 8
  - [ ] JDK 11
  - [ ] JDK 17

### Licensing

- [ ] New dependencies are compatible with the [Apache License 2.0](https://apache.org/licenses/LICENSE-2.0) according to the [License Policy](https://www.apache.org/legal/resolved.html)
- [ ] New dependencies are documented in applicable `LICENSE` and `NOTICE` files

### Documentation

- [ ] Documentation formatting appears as expected in rendered files

thenatog commented on nifi-registry/nifi-registry-core/nifi-registry-web-api/src/main/java/org/apache/nifi/registry/web/security/NiFiRegistrySecurityConfig.java:
```diff
@@ -101,26 +99,17 @@ protected void configure(HttpSecurity http) throws Exception {
-                    .and()
-                .sessionManagement()
-                    .sessionCreationPolicy(SessionCreationPolicy.STATELESS);
+                    .authenticationEntryPoint(http401AuthenticationEntryPoint());
```
This actually looks a little more complex than the authentication entry point I just merged in NiFi. Should the NiFi one contain some of these extra tasks such as logging and setting return content types?

exceptionfactory replied:
Thanks for the note @thenatog. The HTTP Request Logging on NiFi provides HTTP status code details for each URL requested, and Spring Security debug logging could be enabled for additional troubleshooting. With that background, however, it would be worthwhile to consider some of kind of error message string in certain cases. The Authentication Entry Point on NiFi has a few more scenarios to handle, but it sounds worth considering in a follow-on issue.

thenatog commented:
I ran into an issue with this one trying to use LDAP to log in. When I hit the log in button, enter my LDAP creds it, the URL 'https://myhostname.com:18080/nifi-registry-api/access/token/login' returns a 404. The log shows: 2022-07-18 12:49:16,560 INFO [NiFi Registry Web Server-16] o.a.n.r.w.mapper.NotFoundExceptionMapper javax.ws.rs.NotFoundException: HTTP 404 Not Found. Returning Not Found response.

I compared this with main branch and found that I was able to log in using the same LDAP configuration.

exceptionfactory commented:
Thanks for the feedback @thenatog!

The problem was the result of the CSRF checking not being disabled, as it is in the current configuration. I corrected the configuration and rebased, so this is ready for another review.

thenatog commented:
Yeah tested now and it's working. +1 will merge, thanks!
